addons/web_editor/static/lib/odoo-editor/src/commands/toggleList.js

```diff
@@ -26,12 +26,9 @@ HTMLElement.prototype.oToggleList = function (offset, mode = 'UL') {
         restoreCursor();
     } else {
         const list = insertListAfter(this, mode, [this]);
-        for (const attribute of this.attributes) {
-            if (attribute.name === 'class' && attribute.value && list.className) {
-                list.className = `${list.className} ${attribute.value}`;
-            } else {
-                list.setAttribute(attribute.name, attribute.value);
-            }
+        const dir = this.attributes.dir;
+        if (dir) {
+            list.setAttribute('dir', dir.value);
```

Comment:
What made you select `dir` as the only valid attribute to preserve? You say "for `HTMLElement` the existing element will always remain". What do you mean? `HTMLParagraphElement` inherits from `HTMLElement`. Which issue does this fix? I don't see it in the task.

Also:
```suggestion
        if (this.hasAttribute('dir')) {
            list.setAttribute('dir', this.getAttribute('dir'));
```

Reply:
This was not mentioned in the task, I found it in the flow and fixed it.
Here when i used to add a list to `<a href="#">abc[]</a>` it would to convert it to `<ul href="#"><li><a href="#">abc[]</a></li></ul>` which is incorrect behavior.
Also I have rephrased the commit message as it caused confusion.